### ISSUE TITLE
Highlighting Pandoc style citations

### ISSUE THREAD

**Issue by retrography** [marktext/marktext#1316]
### Describe your feature request

It would be great if MarkText would be able to highlight Pandoc-style citations.

### What problem does this feature solve? [optional]

Makes the editor much more adapted to scientific writing.

### Additional context [optional]

Many other Markdown and code editors support that. I am not asking for completion or rendering at this stage. Color-coding is already a great step forward.

If someone points me to the part of the code that must be patched, I may take a stab at it myself.

**Comment by fxha** [comment]
Thank you for taking the time to report this issue! Could you please give an example what you mean by pandoc style citations? I guess you are referring to [this](https://pandoc.org/MANUAL.html#citations) but what exactly?

**Comment by samuelmehr** [comment]
I also would love this feature! What @retrography is describing is a common way of adding citations to academic writing in markdown. Most common format is Better BibTeX, with citations in the format [@CiteKey] (e.g., [@Darwin1859]) and most common manager for references is Zotero.

I posted about this in Typora issues at https://github.com/typora/typora-issues/issues/2880 as it seems that none of the really slick markdown editors yet support this (my Typora issue was closed and merged with other similar feature requests).

Ultimately it would be amazing if MarkText could not only highlight citation keys automatically (as in the screenshots in my Typora issue) but also integrate with `pandoc` and `pandoc-citeproc` to convert citekeys into fully formatted bibliographic references using a citation style (csl) file.

Happy to post more documentation / help to explain how scientists would use and love this feature.

**Comment by samuelmehr** [comment]
This is a common academic workflow: write in markdown, add bibliographic citations in text using Better BibTeX-formatted citation keys (usually generated in Zotero), which often take the format `[@AuthorYear]` as in `[@Mehr2018]`, and then knit the markdown via `pandoc` to replace the citation keys with properly formatted references. So it goes from something that looks like this:

> ## Abstract
> 
> The forms of many animal signals are shaped by their functions, a link arising from production- and reception-related rules that help to maintain reliable signal detection within and across species [@Morton1977; @Endler1993; @Owren2001; @Fitch2002; @Wiley1983; @Krebs1984]. This is especially true of vocalizations, where form-function links have been demonstrated across the animal kingdom in terms of vocalization production [@Owren2001; @Karp2014; @Slaughter2013; @Wagner1989; @Ladich1989] and perception [@Filippi2017; @Lingle2014; @Custance2012; @Magrath2015; @Lea2008]. In humans, an area of importance for effectively transmitting vocal signals is between parents and infants, in both the domains of speech [@Cooper1990; @Fernald1992; @Falk2004] and song [@Trehub1993; @Mehr2016; @Mehr2017b; @Mehr2017; @Mehr2017a; @Kotler2019]. Infant-directed vocalizations are known to differ reliably from adult-directed vocalizations in a few human societies, whether those vocalizations contain speech [@Bryant2007; @Piazza2017; @Broesch2015] or song [@Trehub1993; @Mehr2018; @Mehr2019] — unlike infant-directed vocalizations in most other primates [@Mehr2017a] — but the degree to which their forms and functions are linked is unknown. Here, we show that the acoustic design of infant-directed speech and song shows clear evidence for universality. We collected recordings of infant- and adult-directed singing and speech from 20 human societies (*N* = 1614 recordings from 411 people) and played them to naïve listeners recruited online (*N* = 166,438 ratings from 12,461 people). They reliably distinguished infant-directed vocalizations from adult-directed vocalizations in all societies and identification accuracy was highest, by far, for infant-directed song and infant-directed speech. We then used preregistered exploratory-confirmatory analyses and machine classifiers to determine the acoustic features that reliably characterize infant-directed song and speech across cultures. These included the phonetic space of their formants, their rhythmic structure, and their pitch range attributes, among others, demonstrating universal acoustic differentiation between infant- and adult-directed vocalizations in humans.

to something like this:

<img width="757" alt="Screen Shot 2019-10-24 at 12 31 57 PM" src="https://user-images.githubusercontent.com/41483946/67441686-4a90bd80-f65a-11e9-9cd4-609ee192ca31.png">

**Comment by Jocs** [comment]
@samuelmehr is there another file to store the references? if so, is it saved in the same folder with markdown file.

Does this issue is related to #1358 ？

**Comment by Jocs** [comment]
Maybe also related to #1118

**Comment by samuelmehr** [comment]
Yes, same idea as #1358. All reference data is stored in a `.bib` file that `pandoc` calls on to produce the references.

**Comment by Jocs** [comment]
How can Mark Text find the `.bib` file? is it in the same folder with the markdown file? is there other type file which also used to store these references. Which is most common used?

**Comment by samuelmehr** [comment]
Typically it's stored in the same folder, yes, but you can put it anywhere and specify a path in the md header with
```bibliography: /path/to/bibfile
```

**Comment by Jocs** [comment]
@samuelmehr thanks, I think this is a useful feature for academic use case.

**Comment by samuelmehr** [comment]
it'd be awesome — Typora doesn't support this at all so might also differentiate from them. happy to help beta test if I can. My dev chops are mostly statistical programming, though, so I can't be much help otherwise unfortunately 😬

**Comment by samuelmehr** [comment]
@bwiernik is probably interested in this too!

**Comment by bwiernik** [comment]
Here is a good overview of the syntax options that pandoc supports for citations/bibliographies/references: https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
